src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -311,212 +334,219 @@ void func_80BA8DF4(EnBaba* this, GlobalContext* globalCtx) {
         Math_SmoothStepToS(&this->torsoRot.y, 0, 4, 0x3E8, 1);
     }
 
-    SubS_FillLimbRotTables(globalCtx, this->unk_302, this->unk_326, ARRAY_COUNT(this->unk_302));
+    SubS_FillLimbRotTables(globalCtx, this->limbRotTableY, this->limbRotTableZ, ARRAY_COUNT(this->limbRotTableY));
 
-    if (this->unk_40A & 2) {
-        func_80BA8C90(this, globalCtx);
+    if (this->flags & ENBABA_VISIBLE) {
+        EnBaba_UpdateCollider(this, globalCtx);
     }
 }
 
-s32 func_80BA8F88(EnBaba* this, GlobalContext* globalCtx, ScheduleResult* arg2) {
-    u16 sp26 = (u16)(gSaveContext.save.time - 0x3FFC);
-    u16 temp;
-    u8 sp23 = ENBABA_GET_3F00(&this->actor);
+s32 EnBaba_InitTimePath(EnBaba* this, GlobalContext* globalCtx, ScheduleResult* scheduleResult) {
+    u16 now = SCHEDULE_TIME_NOW;
+    u16 startTime;
+    u8 path = ENBABA_GET_PATH(&this->actor);
 
-    if (D_80BAA4A8[arg2->result] >= 0) {
-        this->unk_410 = SubS_GetAdditionalPath(globalCtx, sp23, D_80BAA4A8[arg2->result]);
+    if (sAdditionalTimePaths[scheduleResult->result] >= 0) {
+        this->timePath = SubS_GetAdditionalPath(globalCtx, path, sAdditionalTimePaths[scheduleResult->result]);
     }
 
-    if (this->unk_410 == NULL) {
+    if (this->timePath == NULL) {
         return false;
     }
 
-    if ((this->unk_434 != 0) && (this->unk_436 >= 0)) {
-        temp = sp26;
+    if ((this->scheduleResult != 0) && (this->timePathTimeSpeed >= 0)) {
```

Comment:
```suggestion
    if ((this->scheduleResult != BABA_SCHEDULE_RESULT_NONE) && (this->timePathTimeSpeed >= 0)) {
```
I find it very unfortunate that `scheduleResult` is the variable name used to represent both `ScheduleResult* scheduleResult` and `u8 scheduleResult` i.e. `scheduleResult.result`. These are two different variables with overlying names (one is the struct itself, one is the return value from running the script), but I haven't thought of anything better. The best thing I've thought of is `this->scheduleResultValue` to indicate that it's the result/return value instead of the `ScheduleResult` struct. Maybe some name needs to change in the schedule system perhaps?

Reply:
Yeah, I thought of that to. Personally I think is should probably be something in the schedule system, but I'm not sure. Probably the struct member `result` needs to change, but I have no idea what to.

Reply:
Perhaps change it to `action` to mirror cutscenes, and how the schedule is communicating what action to take?

Reply:
`action` isn't too bad, especially if it mirrors cutscenes. I'd be curious to see more actors documented though before knowing if `action` is right though. Another name I was thinking of was just `output`?

Reply:
`output` seems fine for now. Will probably add a comment about `action` as a possible name.

Reply:
Actually, I like changing the struct name to `ScheduleOutput` instead.

Reply:
I like that too actually, changing the struct to `ScheduleOutput` and leaving `result as is`. so:
`scheduleOutput.result`
`scheduleOutput.time0`
`scheduleOutput.time1`
`scheduleOutput.hasResult`